activerecord/lib/active_record/insert_all.rb

```diff
@@ -0,0 +1,155 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  class InsertAll
+    attr_reader :model, :connection, :inserts, :on_duplicate, :returning, :unique_by
+
+    def initialize(model, inserts, on_duplicate:, returning: nil, unique_by: nil)
+      @model, @connection, @inserts, @on_duplicate, @returning, @unique_by = model, model.connection, inserts, on_duplicate, returning, unique_by
+      @returning = connection.supports_insert_returning? && primary_keys if @returning.nil?
```

Comment:
The `primary_keys` implementation will always return a "truthy" value:

```
$ be ruby -Iactivesupport/lib -ractive_support/all -e'p Array.wrap(nil)'
[]
```

Is this conditional necessary?  If it is, we should probably add a test and fix the conditional.

Reply:
Oh! My intention here was to set the default value of `@returning` to `primary_keys` if the connection supports RETURNING and to `false` if it doesn't.

So, I guess this is like a compressed version of

```ruby
(connection.supports_insert_returning? ? primary_keys : false)
```

Would it be better to expand it?

Reply:
Ah, I see.  The ternary seems more clear to me.

Reply:
👍 I'll make that change